### PR TITLE
Feature/52 trx page UI and fixed logout 

### PR DESCRIPTION
**Pull request by sadiabbasi**

### CODE REVIEW

File: src/demo/components/TUserInfo.css
Comment (theblockstalk) [inline src/demo/components/TUserInfo.css:1]: empty file. remove or use

File: src/demo/pages/blockchainTx.tsx
Comment (theblockstalk) [inline src/demo/pages/blockchainTx.tsx:1]: use consistent file name "BlockchainTx.tsx"

Comment (theblockstalk) [reply src/demo/pages/blockchainTx.tsx:1]: react component files always have Upper Camel Case file names

File: src/demo/components/TPageHighlighted.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { TH2 } from "../../common/atoms/THeadings";
+import { Highlighter } from "rc-highlight";
+import PropTypes from "prop-types";
+
+const HighlightedPageView = ({ highlighterText, documentLink, githubLink }) => {
+  return (
+    <div className="rightDocs">
```
Comment (theblockstalk) [inline src/demo/components/TPageHighlighted.tsx:7]: What is this class name CSS targeting?

File: src/demo/components/TPageHighlighted.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      <TH2 className="title">Code Snippet</TH2>
+      <div className="highlighter">
```
Comment (theblockstalk) [inline src/demo/components/TPageHighlighted.tsx:9]: CSS target "highlighter"?

File: src/demo/components/TPageHighlighted.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        rel="noreferrer"
+      >
+        View on GitHub
+      </a>
+    </div>
+  );
+};
+
+HighlightedPageView.propTypes = {
+  highlighterText: PropTypes.string,
```
Comment (theblockstalk) [inline src/demo/components/TPageHighlighted.tsx:29]: what is the advantage of PropTypes? why not just `string`?

File: src/demo/components/TUserInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect } from "react";
+import "./TUserInfo.css";
+import { api, ExternalUser, SdkError, SdkErrors } from "@tonomy/tonomy-id-sdk";
+import { useNavigate } from "react-router-dom";
+import { TH3, TP } from "../../common/atoms/THeadings";
+import "@tonomy/tonomy-id-sdk/build/api/tonomy.css";
+import { TButton } from "../../common/atoms/TButton";
+
+export default function TPageLeftView() {
```
Comment (theblockstalk) [inline src/demo/components/TUserInfo.tsx:9]: File name and component name do not match

File: src/demo/components/TUserInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  const [user, setUser] = useState<ExternalUser | null>(null);
+  const [accountName, setAccountName] = useState<string>("");
+  const [username, setUsername] = useState<string>("");
+  const navigation = useNavigate();
+
+  async function onRender() {
+    try {
+      const user = await api.ExternalUser.getUser();
```
Comment (theblockstalk) [inline src/demo/components/TUserInfo.tsx:17]: get this from userStore. Then you do not need to do the handling catches

File: src/demo/components/TUserInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        (e.code === SdkErrors.AccountNotFound ||
+          e.code === SdkErrors.AccountDoesntExist ||
+          e.code === SdkErrors.UserNotLoggedIn)
+      ) {
+        // User not logged in
+        navigation("/");
+        return;
+      }
+
+      console.error(e);
```
Comment (theblockstalk) [inline src/demo/components/TUserInfo.tsx:39]: Use the errorStore for catching errors

File: src/demo/components/TUserInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      alert(e);
+    }
+  }
+
+  useEffect(() => {
```
Comment (theblockstalk) [inline src/demo/components/TUserInfo.tsx:44]: remove, unused

File: src/demo/components/TUserInfo.tsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    <div className="head-subtitle">
+      <TP>You are now logged in with Tonomy ID.</TP>
+      <TP>
+        View your account on the blockchain{" "}
+        <a
+          target={"_blank"}
+          href={
+            "https://local.bloks.io/account/" +
+            accountName +
+            "?nodeUrl=http://localhost:8888"
```
Comment (theblockstalk) [inline src/demo/components/TUserInfo.tsx:57]: this should use settings.blockchainUrl, or something like this. See the logged in page

File: src/demo/layout/main.tsx
Comment (theblockstalk) [inline src/demo/layout/main.tsx:1]: Filename and component name do not match

File: src/demo/layout/main.tsx [hunk 9]
```diff
@@ -12,9 +12,11 @@ import ChatBubbleOutlineOutlinedIcon from "@mui/icons-material/ChatBubbleOutline
 import LogoutIcon from "@mui/icons-material/Logout";
 import logo from "../assets/tonomy-logo48.png";
 import "./main.css";
+import "./pageLayout.css";
 
-const MainLayout = ({ content }) => {
+const MainLayout = ({ onLogout }) => {
```
Comment (theblockstalk) [inline src/demo/layout/main.tsx:17]: Add type to onLogout

File: src/demo/layout/main.css
Comment (theblockstalk) [inline src/demo/layout/main.css:1]: file name should match componet file name, with capitals. MainLayout.css

File: src/demo/layout/main.tsx [hunk 10]
```diff
@@ -12,9 +12,11 @@ import ChatBubbleOutlineOutlinedIcon from "@mui/icons-material/ChatBubbleOutline
...
   const { collapseSidebar } = useProSidebar();
+  const [collapse, setCollapse] = useState(false);
   const [user, setUser] = useState<ExternalUser | null>(null);
   const navigation = useNavigate();
```
Comment (theblockstalk) [inline src/demo/layout/main.tsx:22]: Linie 25. Use `userStore()` instead, and then you do not need to catch errors

Comment (theblockstalk) [reply src/demo/layout/main.tsx:22]: error handling should be with useErrorStore()

File: src/demo/layout/main.tsx [hunk 11]
```diff
@@ -94,15 +89,15 @@ const MainLayout = ({ content }) => {
...
       </div>
-      <div className="main-content">
+      <div className="main-content" style={{ zIndex: collapse ? -1 : 0 }}>
         <Outlet />
       </div>
     </div>
   );
 };
 
 MainLayout.propTypes = {
```
Comment (theblockstalk) [inline src/demo/layout/main.tsx:99]: why not set type in the component function arguments on L17?

File: src/demo/layout/pageLayout.css
Comment (theblockstalk) [inline src/demo/layout/pageLayout.css:1]: match component file name and Upper Camel Case

File: src/demo/components/TPageHighlighted.tsx (hunk 1 above)
Comment (theblockstalk) [reply src/demo/components/TPageHighlighted.tsx:7]: Just found it in pageLayout.css. You should not use the CSS from another components CSS. component CSS should be modular - only used by its component.

If you want common CSS this should be inside a file that is shared by others.

File: src/demo/pages/UserHome.css
Comment (theblockstalk) [inline src/demo/pages/UserHome.css:1]: Why is this deleted?

File: src/demo/pages/blockchainTx.tsx [hunk 12]
```diff
@@ -1,3 +1,107 @@
...
+        </TP>
+        <TP>
+          In this scenario, imagine that you want to sign a blockchain
+          transaction to verify your ownership and authorization of a digital
+          asset.
+        </TP>
+        <TP className="list">
+          Other examples of how to use credentials with Tonomy ID include:
+          <ul>
+            <li>employment contracts</li>
```
Comment (theblockstalk) [inline src/demo/pages/blockchainTx.tsx:35]: Missing the emoji's from the design

File: src/demo/pages/blockchainTx.tsx [hunk 13]
```diff
@@ -1,3 +1,107 @@
...
+              );
+              ...
+            }
+            <button className="tonomy-login-button"
+            onClick={onButtonPress}>
+            Login with {Your Platform Name Here}
+            </button>
+          `}
+        documentLink="https://docs.tonomy.foundation"
+        githubLink="https://github.com/Tonomy-Foundation/Tonomy-App-Websites/blob/master/src/demo/pages/Home.tsx"
```
Comment (theblockstalk) [inline src/demo/pages/blockchainTx.tsx:103]: Link to the correct page

File: src/demo/routes/root.tsx [hunk 14]
```diff
@@ -33,12 +33,14 @@ const AuthRoutes = () => {
...
 const Router: React.FC = () => {
   const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [user, setUser] = useState<ExternalUser | null>(null);
 
   async function onRender() {
     try {
       const user = await api.ExternalUser.getUser();
 
-      if (user) setIsLoggedIn(true);
+      setUser(user);
```
Comment (theblockstalk) [inline src/demo/routes/root.tsx:42]: Should use the userStore

File: src/demo/routes/root.tsx [hunk 15]
```diff
@@ -59,10 +61,21 @@ const Router: React.FC = () => {
...
     onRender();
-  }, []);
+  }, [isLoggedIn]);
+
+  const handleLogout = async () => {
+    try {
+      await user?.logout();
+      setIsLoggedIn(false);
+    } catch (e) {
+      console.error(e);
```
Comment (theblockstalk) [inline src/demo/routes/root.tsx:71]: use the useErrorStore()

File: src/demo/components/TPageHighlighted.tsx (hunk 3 above)
Comment (sadiabbasi) [reply src/demo/components/TPageHighlighted.tsx:29]: because it is already included in React. It's ready to use and has a low learning curve

File: src/demo/components/TUserInfo.tsx (hunk 5 above)
Comment (sadiabbasi) [reply src/demo/components/TUserInfo.tsx:17]: in demo we don't have the userStore can i use it from sso?

File: src/demo/components/TUserInfo.tsx (hunk 6 above)
Comment (sadiabbasi) [reply src/demo/components/TUserInfo.tsx:39]: don't have the error store in demo plus sso.. i guess may be we need to add new task to create errorStore as this is just UI task?

Comment (sadiabbasi) [reply src/demo/components/TUserInfo.tsx:39]: ohh i found that in common

File: src/demo/components/TUserInfo.css [hunk 16]
```diff
@@ -0,0 +1,14 @@
+.head-subtitle {
+    text-align: center;
+    margin-bottom: 20px;
+   
+  }
+
+  .head-subtitle a {
+        color: #0057B2;
```
Comment (theblockstalk) [inline src/demo/components/TUserInfo.css:8]: use theme color instead?

File: src/demo/components/TUserInfo.tsx (hunk 5 above)
Comment (theblockstalk) [reply src/demo/components/TUserInfo.tsx:17]: still not resolved

File: src/demo/layout/MainLayout.css [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+  height: 100vh;
+}
+
+.main-content {
+  position: fixed;
+  right: 0;
+  width: 95%;
+  height: 100%;
+  overflow: auto;
+  /* z-index: -1; */
```
Comment (theblockstalk) [inline src/demo/layout/MainLayout.css:50]: remove comment

File: src/demo/layout/MainLayout.css [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+.ps-sidebar-container , h3{
+  background-image: url("../assets/background.jpeg");
+
+  color: #BEBEBE;
```
Comment (theblockstalk) [inline src/demo/layout/MainLayout.css:56]: can we use theme here?

File: src/demo/layout/MainLayout.tsx [hunk 19]
```diff
@@ -11,58 +10,42 @@ import SwapHorizOutlinedIcon from "@mui/icons-material/SwapHorizOutlined";
...
   const { collapseSidebar } = useProSidebar();
+  const [collapse, setCollapse] = useState(false);
   const [user, setUser] = useState<ExternalUser | null>(null);
   const navigation = useNavigate();
+  const userStore = useUserStore();
 
   async function onRender() {
-    try {
-      const user = await api.ExternalUser.getUser();
+    const user = await api.ExternalUser.getUser();
```
Comment (theblockstalk) [inline src/demo/layout/MainLayout.tsx:28]: instead,
```ts
const user = userStore.user
if (!user) navigate back to login page
```
this will get it from the user store, which is initialized in the login or the callback page not here

File: src/demo/layout/main.css
Comment (theblockstalk) [inline src/demo/layout/main.css:1]: file should be deleted if nothing in it

File: src/demo/pages/PageLayout.css [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+    margin-bottom: 1rem;
+  }
+  
+  .text-header {
+    margin-bottom: 1rem;
+  }
+
+  .nftImage {
+    float: left;
+    background: #CCE9CC;
```
Comment (theblockstalk) [inline src/demo/pages/PageLayout.css:65]: theme colour?

File: src/demo/pages/PageLayout.css [hunk 21]
```diff
@@ -0,0 +1,218 @@
...
+  .nftImageColumn p {
+    line-height: 2.5rem;
+  }
+  
+  .title {
+    background-color: transparent;
+    text-align: left
+  }
+   .title {
+    /* margin-top: auto; */
```
Comment (theblockstalk) [inline src/demo/pages/PageLayout.css:103]: remove comments

File: src/demo/pages/UserHome.tsx [hunk 22]
```diff
@@ -58,7 +58,7 @@ export default function Login() {
 
   return (
     <div className="container">
-      <div className="intro">
+      <div className="pageIntro">
```
Comment (theblockstalk) [inline src/demo/pages/UserHome.tsx:61]: this CSS is coming from which CSS file? again, this should be modular, each file controls its own CSS

File: src/demo/components/TUserInfo.tsx [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+    onRender();
+  }, []);
+  return (
+    <div className="head-subtitle">
+      <TP>You are now logged in with Tonomy ID, as {username}</TP>
+      <TP>
+        View your account on the blockchain{" "}
+        <a
+          target={"_blank"}
+          href={settings.config.blockchainUrl}
```
Comment (theblockstalk) [inline src/demo/components/TUserInfo.tsx:39]:
```html
<a
            target={"_blank"}
            href={
              "https://local.bloks.io/account/" +
              accountName +
              "?nodeUrl=" +
              settings.isProduction() ? settings.blockchainUrl : "http://localhost:8888"
            }
            rel="noreferrer"
          >
```

File: src/demo/pages/UserHome.tsx (hunk 22 above)
Comment (sadiabbasi) [reply src/demo/pages/UserHome.tsx:61]: it's coming from PageLayout.css i made single css file and included it in userHome and blockchain.tsx too

**Comment by theblockstalk** [comment]
ok looks like all that is left is the theme colour, and the shared CSS... everything else fine

**Review by theblockstalk — APPROVED** [review]
